Infrastructure.STS.OrganizationUnit/DomainServices/StsOrganizationUnitService.cs

```diff
@@ -0,0 +1,237 @@
+
+                //TODO: check errors
+
+                var units = listResponse
+                    .ListResponse1
+                    .ListOutput
+                    .FiltreretOejebliksbillede
+                    .Select(snapshot => (new Guid(snapshot.ObjektType.UUIDIdentifikator), snapshot.Registrering.OrderByDescending(x => x.Tidspunkt).FirstOrDefault()))
+                    .Where(x => x.Item2 != null);
+
+                totalResults.AddRange(units);
+
+            } while (currentPage.Count == pageSize);
+
+            // Prepare conversion to import tree
+            var unitsByUuid = totalResults.ToDictionary(unit => unit.Item1);
+            var unitsByParent = totalResults
+                .Where(x => x.Item2.RelationListe.Overordnet != null) // exclude the root
+                .GroupBy(unit => new Guid(unit.Item2.RelationListe.Overordnet.ReferenceID.Item))
+                .ToDictionary(grouping => grouping.Key, grouping => grouping.ToList());
+            var roots = totalResults.Where(x => x.Item2.RelationListe.Overordnet == null).ToList();
+            if (roots.Count != 1)
+            {
+                //TODO: error
+            }
+
+            // Process the tree info from sts org in order to generate the import tree
+            var parentIdToConvertedChildren = new Dictionary<Guid, List<StsOrganizationUnit>>();
+            var idToConvertedChildren = new Dictionary<Guid, StsOrganizationUnit>();
+            var root = roots.Single();
+
+            var processingStack = CreateOrgUnitConversionStack(root, unitsByParent);
+
+            while (processingStack.Any())
+            {
+                var currentUnitUuid = processingStack.Pop();
+                (Guid, RegistreringType1) unit = unitsByUuid[currentUnitUuid];
+
+                var egenskabType = unit.Item2.AttributListe.Egenskab[0];//TODO: Check if we can always depend on this to be there
+                var unitUuid = unit.Item1;
+                var organizationUnit = new StsOrganizationUnit(unitUuid, egenskabType.EnhedNavn, egenskabType.BrugervendtNoegleTekst, parentIdToConvertedChildren.ContainsKey(unitUuid) ? parentIdToConvertedChildren[unitUuid] : new List<StsOrganizationUnit>(0));
+                idToConvertedChildren[organizationUnit.Uuid] = organizationUnit;
+                var parentUnit = unit.Item2.RelationListe.Overordnet;
+                if (parentUnit != null)
+                {
+                    var parentId = new Guid(parentUnit.ReferenceID.Item);
+                    if (!parentIdToConvertedChildren.TryGetValue(parentId, out var parentToChildrenList))
+                    {
+                        parentToChildrenList = new List<StsOrganizationUnit>();
+                        parentIdToConvertedChildren[parentId] = parentToChildrenList;
+                    }
+                    parentToChildrenList.Add(organizationUnit);
+                }
+            }
+
+            return idToConvertedChildren[root.Item1];
```

Comment:
so the purpose of building the tree is in fact only to validate that it is a complete tree with one root.
Maybe the response should just contain the array or both the root and the flat array.